jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -796,25 +796,43 @@ ecma_op_function_get_realm (const ecma_compiled_code_t *bytecode_header_p) /**<
 ecma_global_object_t *
 ecma_op_function_get_function_realm (ecma_object_t *func_obj_p) /**< function object */
 {
-  if (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_FUNCTION)
+  while (true)
   {
-    ecma_extended_object_t *ext_function_obj_p = (ecma_extended_object_t *) func_obj_p;
-    const ecma_compiled_code_t *bytecode_data_p = ecma_op_function_get_compiled_code (ext_function_obj_p);
-    return ecma_op_function_get_realm (bytecode_data_p);
-  }
+    if (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_FUNCTION)
+    {
+      ecma_extended_object_t *ext_function_obj_p = (ecma_extended_object_t *) func_obj_p;
+      const ecma_compiled_code_t *bytecode_data_p = ecma_op_function_get_compiled_code (ext_function_obj_p);
+      return ecma_op_function_get_realm (bytecode_data_p);
+    }
 
-  JERRY_ASSERT (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_NATIVE_FUNCTION);
+    if (ecma_get_object_is_builtin (func_obj_p))
```

Comment:
This check must be inside `if (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_NATIVE_FUNCTION)`. It is strange there was no crash.